src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -3842,6 +3843,14 @@ void G1CollectedHeap::post_evacuate_collection_set(G1EvacuationInfo* evacuation_
 
   evacuation_info->set_collectionset_used_before(collection_set()->bytes_used_before());
   evacuation_info->set_bytes_used(_bytes_used_during_gc);
+
+  start_new_collection_set();
+
+  prepare_tlabs_for_mutator();
+
+  gc_epilogue(false);
```

Comment:
I wonder if `gc_epilogue` could be made more explicit about what it includes, like its neighbors.

Reply:
Not sure what you expect here. The only change (similar to `gc_prologue`) is to remove potentially parallelizable code (TLAB handling), and that it is now part of the "actual collection" scope. I added some comments to indicate what scope in the method is supposed to handle what.